libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,12 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
Also, I would add a comment here about why we need this handler, since it's not obvious.